third-party/llvm/Makefile

```diff
@@ -36,6 +36,17 @@ LLVM_CONFIG_FILE := $(LLVM_INSTALL_DIR)/bin/llvm-config
 LLVM_CLANG_FILE := $(LLVM_INSTALL_DIR)/bin/clang
 LLVM_CLANG_CONFIG_FILE := $(LLVM_INSTALL_DIR)/configured-clang-sysroot-arguments
 
+# Sometimes one is using a custom version of GCC. In that event,
+# we'd like the built clang to use that version's std C++ headers, rather
+# than only finding the versions in /usr/include. To do that, pass
+# GCC_INSTALL_PREFIX for GCC builds.
+CHPL_LLVM_GCC_PREFIX := ""
+ifneq (, $(findstring gnu,$(CHPL_MAKE_HOST_COMPILER)))
+  CHPL_LLVM_GCC_PREFIX_WHICH := $(shell which $(CC))
+  CHPL_LLVM_GCC_PREFIX_DIR := $(subst bin/$(CC),,$(CHPL_LLVM_GCC_PREFIX_WHICH))
+  CHPL_LLVM_GCC_PREFIX := -DGCC_INSTALL_PREFIX=$(CHPL_LLVM_GCC_PREFIX_DIR)
+endif
```

Comment:
I don't think this is going to work on crays. The host compiler will just be gnu and so `CC=gcc` (i.e we don't have to worry about wrapper driver woes), but it looks the install path isn't quite standard. Testing with gcc 6.3 -- `which gcc` is `/opt/gcc/6.3.0/bin/gcc`, but the lib/includes are in `/opt/gcc/6.3.0/snos/` instead of just `/opt/gcc/6.3.0/`

I'm not sure how reliable/stable the `snos` dir is. Do you know what happens if we specify a gcc-install prefix that doesn't exist (either on the current machine or if for example the dir exists where we build a cray-module, but not on the machine the module is being installed on)?

Reply:
This should be resolved in the current version.